### PR TITLE
Implement content viewer UI components

### LINKED ISSUE #29: Implement entry view page
# Overview

The Guide needs to have a way to actually view the stored articles. It should allow the user to easily view the content of an article and provide relevant info about the article. Should also include some way to start an edit suggestion on the article, and a share button.

# Proposal

The components relevant to the content viewer need to be determined and structured in a hierarchy. Then each component will need to be implemented.

## Supplementary Information

image

### LINKED ISSUE #31: [Feature Request] Content Viewer/Header
# Overview
This will display relevant information about the article (contributors, title, last edit date) and house the share and edit buttons.

# Proposal
Relevant components for all the information and interactive elements need to be added so they can be used by the content viewer.

## Supplementary Information
image

### LINKED ISSUE #32: [Feature Request] Header/Share Button
# Overview
A Share Button so readers can easily share the articles.

# Proposal
A component would need to be implemented which will be utilised by the header to place a share button in the header

## Supplementary Information
image

### LINKED ISSUE #33: [Feature Request] Header/Article Information
# Overview
This broadly describes all the elements of the header that will be information about the article. More specifically this includes the title, the contributors, and the last edit date.

# Proposal
The relevant components will need to be implemented and utilised by the header component.

## Supplementary Information
image

### LINKED ISSUE #37: [Feature Request] Content Viewer/Content
# Overview
This would correspond with the actual contents of the article. Displaying them and having the headers be. The user should be able to read the article easily and know the organisation of the article through the headings.

# Proposal
Relevant components will need to be made to display the article contents and the section headings.

## Supplementary Information
image

### PR DESCRIPTION
**Pull request by AvengingPrime**
This closes #31, closes #32, closes #33, closes #37, and other sub-issues that have been merged to the current branch.

### CODE REVIEW

**Comment by AvengingPrime** [comment]
This is what the current components look like when given relevant prop values- 
image


Also, I have added a fonts folder in the styles folder. The fonts folder contains the truetype font files, and a styling file that declares those font faces. I have also added an images folder in the public folder, that currently contains the share image. Other images we use throughout the site (not the ones in articles) will probably go in the same location. So, if the locations of these folders or their organisation needs to change, let me know

File: components/Section.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
```
Comment (JosephChang1) [inline components/Section.js:1]: The name Section.js is also used in my sidebar implementation. Must make sure they don't conflict

File: components/Content.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+import "../styles/Content.css";
+
+export function Content(props)
+{
+    return (
+        <div className = "Content">
+            {props.content.split('\n').map((str) => <p key = {str}>{str}</p>)}
```
Comment (JosephChang1) [inline components/Content.js:8]: str is not defined and we haven't find the way to fetch content from database. We'll think about the detail in the future.

**Review by JosephChang1 — APPROVED** [review]

**Review by rnair9 — APPROVED** [review]
LGTM. Maybe add some documentation throughout.

**Review by WillieCubed — APPROVED** [review]
I've gone ahead and made changes I felt were necessary for the code to be merged into `develop`. Most of them are code style changes and some minor refactoring.

File: components/Section.js (hunk 1 above)
Comment (WillieCubed) [reply components/Section.js:1]: This shouldn't be an issue as the name can be changed when the component is imported. Names are scoped per module (i.e. per file).

File: components/Content.js (hunk 2 above)
Comment (WillieCubed) [reply components/Content.js:8]: In this, `str` is a variable defined by the anonymous function passed to the `map` function. The more verbose version of this is:
```ts
  props.content.split()  // Returns a string[]
    .map(                // Map
      (str: string) => {
        return (
           <p key={str}>
             {str}       // Content passed to render
           </p>
        );
      }
    );
```

I recommend looking at [this page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) for more info on the `map` function. It's very useful!

As for database functionality, we keep that separate from the rest of the components to enforce separation of concerns. These components solely render data, so we'll have to write additional functionality to fetch and transform data from our database.

**Pull request merged**